### PR TITLE
Refactor Enveloper and RestProcessor 

### PR DESCRIPTION
**Pull request by ZeeshanGhalib**

### CODE REVIEW

File: adapters/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/RestProcessor.java [hunk 1]
```diff
@@ -54,14 +61,20 @@ public Response processAsynchronously(final Consumer<JsonEnvelope> consumer,
...
 
-        JsonEnvelope result = function.apply(envelope);
-        Response.ResponseBuilder response =
-                result != null ? Response.status(OK).entity(result.payload().toString()) : Response.status(NOT_FOUND);
-        return response.build();
+        final JsonEnvelope result = function.apply(envelope);
+
+        if (result == null) {
+            LOGGER.error(format("Dispatcher returned a null envelope for %s", envelope.metadata().name()));
+            return Response.status(INTERNAL_SERVER_ERROR).build();
```
Comment (purple52) [inline adapters/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/RestProcessor.java]: Could static import Response.status

File: adapters/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/RestProcessor.java [hunk 2]
```diff
@@ -54,14 +61,20 @@ public Response processAsynchronously(final Consumer<JsonEnvelope> consumer,
...
+        } else if (result.payload() == JsonValue.NULL) {
```
Comment (purple52) [inline adapters/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/RestProcessor.java]: And JsonValue.NULL

File: enveloper/src/main/java/uk/gov/justice/services/event/enveloper/Enveloper.java [hunk 3]
```diff
@@ -56,7 +57,7 @@ void register(@Observes final EventFoundEvent event) {
      * @return a function that wraps objects into an envelope.
      */
     public Function<Object, JsonEnvelope> withMetadataFrom(final JsonEnvelope envelope) {
-        return x -> envelopeFrom(buildMetaData(x, envelope.metadata()), objectToJsonValueConverter.convert(x));
+        return x -> envelopeFrom(buildMetaData(x, envelope.metadata()), x == null ? JsonValue.NULL : objectToJsonValueConverter.convert(x));
```
Comment (purple52) [inline enveloper/src/main/java/uk/gov/justice/services/event/enveloper/Enveloper.java]: Could we do this in the converter? ie, make the converter return NULL if the input is null?

File: adapters/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/RestProcessor.java (hunk 1 above)
Comment (ZeeshanGhalib) [inline adapters/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/RestProcessor.java]: Done.

File: adapters/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/RestProcessor.java (hunk 2 above)
Comment (ZeeshanGhalib) [inline adapters/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/RestProcessor.java]: Done.

File: enveloper/src/main/java/uk/gov/justice/services/event/enveloper/Enveloper.java (hunk 3 above)
Comment (ZeeshanGhalib) [inline enveloper/src/main/java/uk/gov/justice/services/event/enveloper/Enveloper.java]: Done.

**Pull request merged**